examples/rust/custom_space_view/src/color_coordinates_space_view.rs

```diff
@@ -0,0 +1,248 @@
+use re_viewer::external::{
+    egui,
+    re_log_types::EntityPath,
+    re_ui,
+    re_viewer_context::{
+        HoverHighlight, Item, SelectionHighlight, SpaceViewClass, SpaceViewClassName, SpaceViewId,
+        SpaceViewState, TypedScene, ViewerContext,
+    },
+};
+
+use crate::color_coordinates_scene::SceneColorCoordinates;
+
+/// The different modes for displaying color coordinates in the custom space view.
+#[derive(Default, Debug, PartialEq, Clone, Copy)]
+enum ColorCoordinatesMode {
+    #[default]
+    Hs,
+    Hv,
+    Rg,
+}
+
+impl std::fmt::Display for ColorCoordinatesMode {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match self {
+            ColorCoordinatesMode::Hs => "Hue/Saturation".fmt(f),
+            ColorCoordinatesMode::Hv => "Hue/Value".fmt(f),
+            ColorCoordinatesMode::Rg => "Red/Green".fmt(f),
+        }
+    }
+}
+
+/// Space view state for the custom space view.
+///
+/// This state is preserved between frames, but not across Viewer sessions.
+#[derive(Default)]
+pub struct ColorCoordinatesSpaceViewState {
+    // TODO(wumpf/jleibs): This should be part of the Blueprint so that it is serialized out.
+    //                      but right now there is no way of doing that.
+    mode: ColorCoordinatesMode,
+}
+
+impl SpaceViewState for ColorCoordinatesSpaceViewState {
+    fn as_any(&self) -> &dyn std::any::Any {
+        self
+    }
+
+    fn as_any_mut(&mut self) -> &mut dyn std::any::Any {
+        self
+    }
+}
+
+#[derive(Default)]
+pub struct ColorCoordinatesSpaceView;
+
+impl SpaceViewClass for ColorCoordinatesSpaceView {
+    // State type as described above.
+    type State = ColorCoordinatesSpaceViewState;
+
+    // Scene context is shared between all scene parts.
+    // For this Space View we don't need any scene context.
+    type Context = ();
+
+    // Collection of scene parts that are needed to display a frame.
+    // We only have a single scene part, but it could be a collection of various.
+    type SceneParts = SceneColorCoordinates;
+
+    // Scene parts can have a common data object that they expose.
+    // For this Space View this is not needed.
+    type ScenePartData = ();
+
+    fn name(&self) -> SpaceViewClassName {
+        // Name and identifier of this Space View.
+        "Color Coordinates".into()
```

Comment:
not for this PR, but I think we should split this. Having the user-visible name also be the unique identifier isn't great. If one wants to rename the space view class we would need to also update the blueprint code. Not to speak of localization.

I think it would make sense with something like `ext.color_coordinates` as the unique identifier and "Color Coordinates" as the name, and then have `rerun.spatial2d` etc for the built-in space view classes

Reply:
https://github.com/rerun-io/rerun/issues/2336